### ISSUE TITLE
Odoo 11+ incompatible with new Python Versions (3.8+) due to requirements.txt

### ISSUE THREAD

**Issue by Armitxes** [odoo/odoo#41313]
Impacted versions: All new Python 3.5+ Versions (11, 12, 13, etc.)

Steps to reproduce: Install python 3.8.0 or higher

Current behavior: 
```
python -m pip install -r /opt/odoo/12.0/base/odoo/community/requirements.txt
...
psycopg/psycopgmodule.c: In function ‘psyco_is_main_interp’:
psycopg/psycopgmodule.c:689:18: error: dereferencing pointer to incomplete type ‘PyInterpreterState’ {aka ‘struct _is’}
     while (interp->next)
```

Expected behavior: No issues at all, as all python versions >=3.5 are supported according to docs

Solution:
The issue is solved by psycopg2 (see https://github.com/psycopg/psycopg2/issues/854 or https://github.com/psycopg/psycopg2/commit/c929f200484f5a2355edf769aaaee1e036129fd8), but in odoos requirements.txt we have:
```
psycopg2==2.7.3.1; sys_platform != 'win32'
psycopg2==2.8.3; sys_platform == 'win32'
```
These versions are simply too old and need to be upgraded to atleast 2.8.4

**Comment by pedrobaeza** [comment]
@d-fence I would say that official documentation should point that P3.8 is not compatible.

**Comment by Armitxes** [comment]
Discovered more problems.

Python 3.8 removed long-time deprecated commands such as
`time.clock()` yet odoo and other package versions in the requirements.txt still require some of these deprecated commands.

Might be best to do as @pedrobaeza suggested and to mention Python 3.8+ as not supported in the docs.

**Comment by gustavovalverde** [comment]
Landed here. Supporting @pedrobaeza suggestion if this fixes can't land stable.

**Comment by mart-e** [comment]
Hello,

Short answer: Odoo 11.0 supports 3.8 but requirements.txt does not, and it is ok.

We have fixed the requirements.txt at #42705 in 13.0 but we do not plan to backport this. Python 3.8 is still not installed by default on most mainstream distributions (some like Archlinux already made the switch but this is not the norm) and is then a manual process. If you are living on the edge, tweaking the `requirements.txt` to use different versions of packages should not be an issue for you (or life on the edge with Odoo too and upgrade to 13.0).

The main incompatibilitis of 3.8 were fixed at #40406
`time.clock()` is [in a test module](https://github.com/odoo/odoo/blob/24d110f14d0a1137e6d1e6377845de37614fea26/odoo/addons/test_limits/models.py#L38) so should not matter in production but good to fix nevertheless (not sure which version), thanks for the notice.

If you see other incompatible code in business code, don't hesitate to let us know but I don't think it is the case (I am myself using python 3.8 for development without issues).
